openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -611,6 +613,39 @@ def post(self, request):
                     log.exception('Error sending out deletion notification email')
                     raise exc
 
+                # Re-use Ecommerce API client factory to build an API client for Commerce Coordinator...
+                service_user = get_user_model().objects.get(
+                    username=settings.COMMERCE_COORDINATOR_SERVICE_WORKER_USERNAME
+                )
+                api_client = get_ecommerce_api_client(service_user)
+                retirement_url = urljoin(
+                    settings.COMMERCE_COORDINATOR_URL_ROOT,
+                    settings.COMMERCE_COORDINATOR_USER_RETIREMENT_PATH
+                )
+
+                lms_user_id = request.user.id
+
+                try:
+                    api_client.post(
+                        retirement_url,
+                        json={
+                            'edx_lms_user_id': lms_user_id,
+
+                        }
+                    ).raise_for_status()
+                except Exception as exc:  # pylint: disable=broad-except
+                    # Catch any possible exceptions from the Commerce Coordinator service to ensure we fail gracefully
+                    log.exception(
+                        "Unexpected exception while attempting to retire Commercetools user in Coordinator [%s], "
+                        "user ID [%s]: ",
+                        lms_user_id,
+                        str(exc)
+                    )
+
+                log.info("User [%s] successfully sent to Commerce Coordinator for retirement in Commercetools.",
+                         lms_user_id
+                        )
```

Comment:
I'm a little worried about references to CommerceTools in OpenSource code. Is this CommerceTools specific? Or can this messaging be generalized?

Reply:
I can generalize the messaging here. The goal of the post to Coordinator is to connect to Commercetools and retire the Commercetools customer object with the same lms_user_id if one exists. Will generalizing the message to simply say we're retiring the user in Coordinator be enough, or is that still too sensitive?

Reply:
I'm not terribly familiar with Commerce Coordinator? Is this part of Open edX? Or is this a service specific to 2U?

Reply:
It's specific to 2U. Based on @deborahgu's comment, where this proposed functionality lives will probably be changed